src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java

```diff
@@ -1570,7 +1571,147 @@ public String getCriteriaSql(ObservationPeriod criteria)
     
     return query;
   }
-
+  
+  @Override
+  public String getCriteriaSql(PayerPlanPeriod criteria)
+  {
+	String query = PAYER_PLAN_PERIOD_TEMPLATE;
+	
+	String startDateExpression = "C.payer_plan_period_start_date";
+	String endDateExpression = "C.payer_plan_period_end_date";
+	
+	ArrayList<String> joinClauses = new ArrayList<>();
+	
+	if (criteria.ageAtStart != null || criteria.ageAtEnd != null || (criteria.gender != null && criteria.gender.length > 0))
+	  joinClauses.add("JOIN @cdm_database_schema.PERSON P on C.person_id = P.person_id");
+	
+	query = StringUtils.replace(query, "@joinClause", StringUtils.join(joinClauses, "\n"));
+	
+	ArrayList<String> whereClauses = new ArrayList<>();
+	
+	//first
+	if (criteria.first != null && criteria.first == true)
```

Comment:
I think `... && criteria.first` is enough

Reply:
i think this is inline with other patterns. 
e.g.
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L633
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L717
https://github.com/OHDSI/circe-be/blob/6b602fd0b405e0cfe0846e9189b613888326ab72/src/main/java/org/ohdsi/circe/cohortdefinition/CohortExpressionQueryBuilder.java#L879

etc